### PR TITLE
fix(card): prevent header from overlapping with checkbox

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #4802 

## Summary

This adds some padding to the header when the card is selectable to prevent overlap with the checkbox.

### CODE REVIEW

File: src/components/card/card.scss [hunk 1]
```diff
@@ -69,6 +69,13 @@
     font-medium;
 }
 
+:host([selectable]) {
+  .header {
+    // prevents overlap with checkbox
+    padding-inline-end: calc(theme("spacing.3") + theme("spacing.5"));
```
Comment (anveshmekala) [inline src/components/card/card.scss:75]: Nitpick : can we replace `calc` with `theme("spacing[8.5]")`

**Review by anveshmekala — APPROVED** [review]
🚀

File: src/components/card/card.scss (hunk 1 above)
Comment (jcfranco) [reply src/components/card/card.scss:75]: Yeah, I think I'll replace that with a comment to explain that it's the heading + no-overlap spacing.

File: src/components/card/card.stories.ts [hunk 2]
```diff
@@ -235,6 +235,21 @@ export const thumbnailRounded = (): string => html`
   </div>
 `;
 
+export const headerDoesNotOverlapWithCheckbox_TestOnly = html`
```
Comment (benelan) [inline src/components/card/card.stories.ts:238]: Take 3:

```js
export const headerDoesNotOverlapWithCheckbox_TestOnl = (): string => html`
```

Comment (jcfranco) [reply src/components/card/card.stories.ts:238]: Thanks for catching this.

**Pull request merged**